rdkit/DataStructs/TopNContainer.py

```diff
@@ -58,7 +58,10 @@ def GetExtras(self):
     return self.extras
 
   def __len__(self):
-    return self._size
+    if self._size >= 0:
+      return self._size
+    else:
+      return len(self.best)
```

Comment:
The case of a negative size (= keep all elements) was not handled in this function.